Added pyswmm simulation class; restructured and added descriptors

bemcdonnell opened the pull request:
@goanpeca, I made some pretty significant changes here.  Please review the new pyswmm.Simulation class and Links.py link classes. Descriptors have been incorporated to *set* different parameters. 

I updated all of the testing files as well so I'm sorry for all the noise

bemcdonnell commented on test/TestModel10_simulationItermethods.py:
```diff
@@ -0,0 +1,42 @@
+'''
```
@michaeltryby,  The following illustrates three approaches to run a simulation with pyswmm.

goanpeca commented on pyswmm/pyswmm.py:
```diff
@@ -0,0 +1,156 @@
+"""Base class for a SWMM Simulation.
```
"""Base class for a SWMM Simulation."""

goanpeca commented on pyswmm/pyswmm.py:
```diff
@@ -0,0 +1,156 @@
+__author__ = 'Bryant E. McDonnell (EmNet LLC) - bemcdonnell@gmail.com'
+__copyright__ = 'Copyright (c) 2016 Bryant E. McDonnell'
+__licence__ = 'BSD2'
+__version__ = '0.2.1'
```
All these variables should be moved to the `pyswmm/__init__.py`

goanpeca commented on pyswmm/pyswmm.py:
```diff
@@ -0,0 +1,156 @@
+        self._model.swmm_end()
+        self._model.swmm_close()
+
+    def step_advance(self,advance_seconds):
```
`self, advance_seconds`  space

goanpeca commented on pyswmm/pyswmm.py:
```diff
@@ -0,0 +1,156 @@
+        """
+
+    def close(self):
+        self.__exit__()
```
no docstring?

goanpeca commented on test/TestModel1_CurrentTime.py:
```diff
@@ -13,10 +13,10 @@
 sys.path.append(os.getcwd()+'\\..\\pyswmm\\')
 
-from swmm5 import pyswmm
+from swmm5 import PYSWMM
```
The object should probably be called

`PySWMM`

goanpeca commented on pyswmm/links.py:
```diff
@@ -0,0 +1,693 @@
+__copyright__ = 'Copyright (c) 2016 Bryant E. McDonnell'
+__licence__ = 'BSD2'
+__version__ = '0.2.1'
+
```
All this should go into the `pyswmm/__init__.py`

goanpeca commented on pyswmm/links.py:
```diff
@@ -0,0 +1,693 @@
+            return linkobject
+        else:
+            raise StopIteration()        
+    @property
```
space between lines

goanpeca commented on pyswmm/links.py:
```diff
@@ -0,0 +1,693 @@
+            raise StopIteration()        
+    @property
+    def linkid(self):
+        "Link ID"
```
"""Link ID."""

goanpeca commented on pyswmm/links.py:
```diff
@@ -0,0 +1,693 @@
+    @property
+    def linkid(self):
+        "Link ID"
+        return self._model.getObjectId(ObjectType.LINK,self._cuindex)
```
`return self._model.getObjectId(ObjectType.LINK, self._cuindex)`

goanpeca commented on pyswmm/links.py:
```diff
@@ -0,0 +1,693 @@
+        self._model.setLinkParam(self._linkid,LinkParams.q0, param)
+    
+    @property
+    def qlimit(self):
```
these names are weird :-|

limit.. q0 ... offset1 etc....

I think we can use much better names,

goanpeca commented on pyswmm/links.py:
```diff
@@ -0,0 +1,693 @@
+        return self._model.getLinkParam(self._linkid,LinkParams.cLossAvg)
+
+    @clossave.setter
+    def clossave(self, param):
```
these names are hard to understand...

`average_head_loss` is already more clear than `clossave`

goanpeca commented on pyswmm/links.py:
```diff
@@ -0,0 +1,693 @@
+        self._model.setLinkParam(self._linkid,LinkParams.seepRate, param)
+        
+    @property
+    def flow(self):
```
For instance, flow :-), nice clear name!

goanpeca commented on pyswmm/links.py:
```diff
@@ -0,0 +1,693 @@
+        return self._model.getLinkResult(self._linkid,LinkResults.newFlow)
+    
+    @property
+    def depth(self):
```
awesome name!

goanpeca commented on pyswmm/links.py:
```diff
@@ -0,0 +1,693 @@
+        return self._model.getLinkResult(self._linkid,LinkResults.newVolume)
+    
+    @property
+    def usxsectionarea(self):
```
`ups_xsection_area` ? we can go over what names and conventiones should we use to make them short enough but understandable enough

goanpeca commented on pyswmm/links.py:
```diff
@@ -0,0 +1,693 @@
+        return self._model.getLinkResult(self._linkid,LinkResults.surfArea2)
+    
+    @property
+    def currentsetting(self):
```
current_setting

goanpeca commented on pyswmm/links.py:
```diff
@@ -0,0 +1,693 @@
+        return self._model.getLinkResult(self._linkid,LinkResults.setting)
+    
+    @property
+    def targetsetting(self):
```
target_setting

bemcdonnell commented:
@goanpeca I addressed all the comments.  For sake of moving this project forward, I am going to go ahead and merge these changes over and i'll cascade them back up to the master branch.
